homeassistant/helpers/translation.py

```diff
@@ -135,17 +135,17 @@ def _merge_resources(
 
 
 def build_resources(
-    translation_strings: dict[str, dict[str, Any]],
+    translation_strings: dict[str, dict[str, dict[str, Any] | str]],
     components: set[str],
     category: str,
 ) -> dict[str, dict[str, Any] | str]:
     """Build the resources response for the given components."""
     # Build response
     return {
-        component: translation_strings[component][category]
+        component: category_strings
         for component in components
-        if category in translation_strings[component]
-        and translation_strings[component][category] is not None
+        if (component_strings := translation_strings.get(component))
+        and (category_strings := component_strings.get(category))
```

Comment:
This changes the semantics somewhat since we now check for truthiness instead of for not None. I'm not sure if that matters?

Reply:
We would get KeyError if the integration had failed to load and we skipped it

Reply:
OK